specs/experimental/fault-proof/stage-one/bond-incentives.md

```diff
@@ -58,3 +64,79 @@ are located on the same level as honest claims, but situated to its right (i.e.
+
+## Fault Proof Mainnet Incentives
+
+This section describes the specific bond incentives to be used for the Fault Proof Mainnet launch of the OP Stack fault
+proof system.
+
+### Gas Price Assumption
+
+FPM bonds are to assume a fixed 200 Gwei gas price.
+Future iterations of the fault proof may include a dynamic gas price calculation.
+For the moment, we suppose that the `GUARDIAN` address may account for increased average gas prices by updating the
+`OptimismPortal` contract to a new respected game type with a higher assumed gas price.
+
+### Bond Scaling
+
+FPM bonds are priced in the amount of gas that they are intended to cover.
+Bonds start at the very first depth of the game at a baseline of `400_000` gas.
+The `400_000` value is chosen as a deterrence amount that is approximately double the cost to respond at the top level.
+Bonds scale up to a value of `200_000_000` gas, a value chosen to cover approximately double the cost of a max-size
+Large Preimage Proposal.
+
+We use a multiplicative scaling mechanism to guarantee that the ratio between bonds remains constant.
+We determine the multiplier based on the proposed `MAX_DEPTH` of 73.
+We can use the formula `x = (200_000_000 / 400_000) ** (1 / 73)` to determine that `x = 1.089`.
+At each depth `N`, the amount of gas charged is therefore `400_000 * (1.089 ** N)`
+
+### Required Bond Formula
+
+Applying the [Gas Price Assumption](#gas-price-assumption) and [Bond Scaling](#bond-scaling) specifications, we have a
+`getRequiredBond` function:
+
+```python
+def get_required_bond(position):
+    assumed_gas_price = 200 gwei
+    base_gas_charged = 400_000
+    gas_charged = 400_000 * (1.089 ** position.depth)
+    return gas_charged * assumed_gas_price
+```
+
+### DelayedWETH
+
+FPM introduces a contract `DelayedWETH` designed to hold the bonded ETH for each
+[Fault Dispute Game](./fault-dispute-game.md).
+`DelayedWETH` is an extended version of the standard `WETH` contract that introduces a delayed unwrap mechanism that
+allows an owner address to function as a backstop in the case that a Fault Dispute Game would
+incorrectly distribute bonds.
+
+`DelayedWETH` is modified from `WETH` as follows:
+
+- `DelayedWETH` is an upgradeable proxy contract.
+- `DelayedWETH` has an `owner()` address
+- `DelayedWETH` has a `delay()` function that returns a period of time that withdrawals will be delayed.
+- `DelayedWETH` has an `unlock()` function that sets a timer within a mapping called `withdrawals` keyed by address. The
+timer for the caller address is set to the current timestamp whenever `unlock()` is called. `unlock()` will not be
+callable when `SuperchainConfig.paused()` is `true`.
+- `DelayedWETH` modifies the `WETH.withdraw` function such that an address cannot withdraw unless their timer value
+within the `withdrawals` mapping is non-zero and the value is more than `delay()` seconds ago. `withdraw()` will not be
+callable when `SuperchainConfig.paused()` is `true`.
```

Comment:
This is confusing it seems like the value is set to the timestamp but then it must be more than `delay()` "seconds ago". It would make more sense to me to have the `unlock()` function set the future timestamp at which the unlock occurs and then the `withdraw` function could just check if `block.timestamp > DelayedWETH.withdrawals(address(this))`?

Reply:
Ah I understand the calculation that is intended to be performed here. The `withdrawal` function would just check if `block.timestamp > withdrawals[address(fdg)] + delay()`...

Reply:
It would be even more precise to change `"timer value"` to something like `"withdrawal unlock timestamp"` or something of the sort

Reply:
Good callout, will change this. Idea is that it makes it easy to change the delay period globally if necessary.